etc/kubernetes/helm/templates/deployment.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: enterprise-gateway
+  namespace: enterprise-gateway
+  labels:
+    gateway-selector: enterprise-gateway
+    app: enterprise-gateway
+    component: enterprise-gateway
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      gateway-selector: enterprise-gateway
+  template:
+    metadata:
+      labels:
+        gateway-selector: enterprise-gateway
+        app: enterprise-gateway
+        component: enterprise-gateway
+    spec:
+      # Created by this chart.
+      serviceAccountName: enterprise-gateway-sa
+      containers:
+        - name: enterprise-gateway
+          image: {{ .Values.image }}
+          imagePullPolicy: {{ .Values.imagePullPolicy }}
+          args: ["--gateway"]
+          env:
+          - name: EG_NAMESPACE
+            value: {{ .Release.Namespace }}
+          - name: EG_KERNEL_CLUSTER_ROLE
+            value: {{ .Values.kernel_cluster_role }}
+          - name: EG_SHARED_NAMESPACE
+            value: {{ if .Values.shared_namespace }}"True"{{ else }}"False"{{ end }}
+          - name: EG_MIRROR_WORKING_DIRS
+            value: {{ if .Values.mirror_working_dirs }}"True"{{ else }}"False"{{ end }}
+          - name: EG_CULL_IDLE_TIMEOUT
+            value: !!str {{ .Values.cull_idle_timeout }}
+          - name: EG_LOG_LEVEL
+            value: {{ .Values.log_level }}
+          - name: EG_KERNEL_LAUNCH_TIMEOUT
+            value: !!str {{ .Values.kernel_launch_timeout }}
+          - name: EG_KERNEL_WHITELIST
+            value: {{ toJson .Values.kernel_whitelist | squote }}
+          ports:
+          - containerPort: 8888
+{{ if .Values.nfs.enabled }}
+          volumeMounts:
+          - name: kernelspecs
+            mountPath: "/usr/local/share/jupyter/kernels"
+        volumes:
+        - name: kernelspecs
+          nfs:
+            server: {{ .Values.internal_server_ip_address }}
+            path: "/usr/local/share/jupyter/kernels"
+{{ end }}
```

Comment:
I'm trying to enable the nfs portion and getting the following ...
```
helm upgrade enterprise-gateway etc/kubernetes/helm
UPGRADE FAILED
ROLLING BACK
Error: YAML parse error on enterprise-gateway/templates/deployment.yaml: error converting YAML to JSON: yaml: line 54: did not find expected '-' indicator
```
I can't figure out what's going on since this matches the previous yaml.  Any ideas?

Reply:
Just discovered `helm lint` and here's what it produces...
```
helm lint etc/kubernetes/helm
==> Linting etc/kubernetes/helm
[ERROR] Chart.yaml: directory name (helm) and chart name (enterprise-gateway) must be the same
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/deployment.yaml: unable to parse YAML
	error converting YAML to JSON: yaml: line 54: did not find expected '-' indicator
```
I then renamed the helm directory to enrterprise-gateway (per the error) and it went away.  This parsing error only shows when nfs.enabled is true.

Reply:
It might be the missing `nfs.` prefix in `internal_server_ip_address` in the template that you just discovered.. A missing value may have tripped up the parser. Please give it another shot with the fix and let me know if it works! (I don't have an NFS server here, although if needed I can probably repro the issue even without one.)

Reply:
I had already been running that way.  Since this can reproduce with `helm lint` I don't think NFS is required.  I'm happy to check it once we get past this parse issue.  I've even tried removing the conditional and still get the same thing.

Reply:
Strangely, I don't get a repro with `helm lint`, but I do get one with a deploy using `--set nfs.enabled=true`.. Looking into it.

Reply:
Found the issue.  The `volumes` block needs to be two spaces to the left.  So the indentation was throwing things off.

Reply:
Yes, just found that as well! Fix tested and pushed.